etcdserver/raft.go

```diff
@@ -223,21 +221,38 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 				r.raftStorage.Append(rd.Entries)
 
 				if !islead {
+					// signal msgSnapC if 'MsgSnap' exists
```

Comment:
finish processing incoming messages before we signal raftdone chan.

i want to make the comment more generic in case we change the impl.